### PR TITLE
Added: WPML "domain per language" compatibility.

### PR DESCRIPTION
**Pull request by Dan0sz**
This PR adds native compatibility with WPML's "a different per language" compatibility (which currently isn't available for Multisite) by adding a **Language Domains** pulldown menu, which allows the user to map each configured Language Domain in WPML to a Plausible Analytics dashboard:

afbeelding

Like in previous versions, the default value of the Domain Name for each Language Domain is the same value of the selected Language Domain:

afbeelding

The Create Token link and Connect button behave the same as they did before UX-wise. This PR also fixes an issue where the "Connect" button was sometimes mislabeled as "Connected".

The configuration wizard remains unchanged to not overcomplicate it. If a new Plausible user configured WPML to use its language per domain feature, the user will be notified of the new feature:

afbeelding

The same goes for users right after updating:

afbeelding

For every configured Language Domain it adds a link to the Admin Bar menu when "View stats in WordPress" is enabled:

afbeelding

It also adds a filter for the Admin Bar menu that allows devs to filter out items/links to stats they don't want/need: `plausible_analytics_admin_bar_view_analytics`.

## Summary by CodeRabbit

* **New Features**
  * Added multilingual “Language Domains” setup with per-language connection credentials and connect/connected states.
  * Added a dismissible multilingual admin notice with Ajax-based dismissal.
* **Bug Fixes**
  * Improved domain/API token normalization and validation across per-language credential sets.
  * Updated shared-link selection and refined connect-button/spinner feedback for more reliable setup.
* **Refactor**
  * Provisioning, tracker configuration updates, and cron downloads now operate per configured API token/domain.

### CODE REVIEW

File: src/Admin/Actions.php [hunk 1]
```diff
@@ -51,38 +97,56 @@ public function register_assets( $current_page ) {
...
 		}
+
+		$settings_url = admin_url( 'options-general.php?page=plausible_analytics' );
+		?>
+		<div id="message" class="notice notice-info is-dismissible plausible-analytics-multilang-notice">
+			<p>
+				<?php printf(
+				/* translators: 1: Multilang plugin name (e.g. WPML), 2: URL to settings page. */
+					__( 'Plausible Analytics now supports %1$s! Connect each domain in %1$s to its own Plausible Analytics dashboard on the <a href="%2$s">settings screen</a>.', 'plausible-analytics' ),
+					$multilang_plugin,
```
[the remark this answers is not in the document]
Comment (Dan0sz) [reply src/Admin/Actions.php:145]: @coderabbitai This feature was non-existent before this PR. So, it's safe to assume that every language domain isn't mapped. Users should see the banner after updating, and can dismiss it themselves. This is a non-issue, IMO.

File: src/Admin/Provisioning.php [hunk 2]
```diff
@@ -118,122 +116,89 @@ private function init() {
...
 
-		if ( $type === 'Revenue' ) {
-			$props['goal']['currency'] = $currency;
-		}
+			$client = ( new ClientFactory() )->build();
 
-		if ( $type === 'Pageview' ) {
-			unset( $props['goal']['event_name'] );
+			if ( $client instanceof Client && $client->validate_api_token() ) {
+				$clients[ $key ] = $client;
```
[the remark this answers is not in the document]
Comment (Dan0sz) [reply src/Admin/Provisioning.php]: @coderabbitai Each token is comprised of 40 unique characters and a "plausible-plugin-" prefix. The chance that 2 tokens are the same is virtually non-existent. This is a non-issue, IMO.

File: src/Ajax.php [hunk 3]
```diff
@@ -402,6 +323,11 @@ function ( $option ) {
 
 		Messages::set_success( __( 'Settings saved.', 'plausible-analytics' ) );
 
+		/**
+		 * Allow devs to perform additional actions.
+		 */
+		do_action( 'plausible_analytics_settings_saved', $settings );
```
[the remark this answers is not in the document]
Comment (Dan0sz) [reply src/Ajax.php:336]: @coderabbitai it's now in line with the same action which is triggered in another place. What I supposed we could do is trigger a doing_it_wrong() warning for people using that filter?

File: src/Client.php [hunk 4]
```diff
@@ -331,84 +251,187 @@ public function bulk_create_shared_links() {
...
+	 */
+	public function is_api_token_valid() {
+		$token        = $this->api_instance->getConfig()->getPassword();
+		$valid_tokens = get_transient( 'plausible_analytics_valid_token' );
+
+		if ( ! is_array( $valid_tokens ) ) {
+			return false;
+		}
+
+		return isset( $valid_tokens[ $token ] ) && $valid_tokens[ $token ] === true;
```
[the remark this answers is not in the document]
Comment (Dan0sz) [reply src/Client.php:434]: @coderabbitai This is the same non-issue I reported earlier. An API token consists of a "plausible-plugin-" prefix followed by 40 unique characters. The chance that 2 API tokens end up being the same is virtually non-existent.
